src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/compute/v4_9/CloudStackComputePlugin.java

```diff
@@ -107,165 +107,195 @@ public String requestInstance(ComputeOrder computeOrder, CloudStackUser cloudUse
                 .userData(userData)
                 .networksId(networksId)
                 .build(this.cloudStackUrl);
+        DeployVirtualMachineResponse deployVirtualMachineResponse = doRequestInstance(
+                deployVirtualMachineRequest, cloudUser);
 
-        CloudStackUrlUtil.sign(request.getUriBuilder(), cloudUser.getToken());
-
-        String jsonResponse = null;
-        try {
-            jsonResponse = this.client.doGetRequest(request.getUriBuilder().toString(), cloudUser);
-
-            synchronized (computeOrder) {
-                ComputeAllocation actualAllocation = new ComputeAllocation(
-                        serviceOffering.getCpuNumber(),
-                        serviceOffering.getMemory(),
-                        1,
-                        diskOffering.getDiskSize());
-                computeOrder.setActualAllocation(actualAllocation);
-            }
-        } catch (HttpResponseException e) {
-            CloudStackHttpToFogbowExceptionMapper.map(e);
-        }
-
-        DeployVirtualMachineResponse response = DeployVirtualMachineResponse.fromJson(jsonResponse);
-
-        return response.getId();
+        updateComputeOrder(computeOrder, serviceOffering, diskOffering);
+        return deployVirtualMachineResponse.getId();
     }
 
     @Override
-    public ComputeInstance getInstance(ComputeOrder order, CloudStackUser cloudUser) throws FogbowException {
-        GetVirtualMachineRequest request = new GetVirtualMachineRequest.Builder()
+    public ComputeInstance getInstance(@NotNull final ComputeOrder order,
+                                       final CloudStackUser cloudUser)
+            throws FogbowException {
+
+        checkCloudUser(cloudUser);
+        GetVirtualMachineRequest getVirtualMachineRequest = new GetVirtualMachineRequest.Builder()
                 .id(order.getInstanceId())
                 .build(this.cloudStackUrl);
-
-        CloudStackUrlUtil.sign(request.getUriBuilder(), cloudUser.getToken());
-
-        String jsonResponse = null;
-        try {
-            jsonResponse = this.client.doGetRequest(request.getUriBuilder().toString(), cloudUser);
-        } catch (HttpResponseException e) {
-            CloudStackHttpToFogbowExceptionMapper.map(e);
-        }
-
-        GetVirtualMachineResponse computeResponse = GetVirtualMachineResponse.fromJson(jsonResponse);
-        List<GetVirtualMachineResponse.VirtualMachine> vms = computeResponse.getVirtualMachines();
-        if (vms != null) {
-            return getComputeInstance(vms.get(0), cloudUser);
-        } else {
-            throw new InstanceNotFoundException();
-        }
+        GetVirtualMachineResponse getVirtualMachineResponse = doGetInstance(getVirtualMachineRequest, cloudUser);
```

Comment:
The _doGetInstance_ method could return a ComputeInstance directly.

Reply:
I will fix this to follow the Fogbow PatternI will fix this to follow the Fogbow Pattern